mautic/mautic
# Segments filters dictionary improvements

This PR is rebase of #7789 for 3.x branch

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | N
| New feature? | Y 
| Automated tests included? | N
| Related user documentation PR URL | N
| Related developer documentation PR URL | https://github.com/mautic/mautic/pull/7789
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
This commit enables two events.
One for feeding the filters lists from event listeners.
Second event for filters evaluation.

[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1. 
2. 

#### Steps to test this PR:
This commit should have no influence on anything. It means that no visible testable change will be.
Check that filters in segment edit is still the same as before checking out this PR.
Check that filters in dynamic contents are the same.
Check dynamic content filters do work correctly.

#### List deprecations along with the new alternative:
1. 
2. 

#### List backwards compatibility breaks:
1. 
2.
